modules/health-check/health-check-unary/src/main/scala/higherkindness/mu/rpc/healthcheck/service.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.rpc.healthcheck
+
+import higherkindness.mu.rpc.protocol.{service, Empty, Protobuf}
+object service {
+
+  @service(Protobuf)
+  trait HealthCheckServiceUnary[F[_]] {
```

Comment:
I'd change the health service by:

* Use the `Ref.update` (instead of `Ref.tryUpdate`) in order to remove the `WentNice` object and return just a `Unit`
* Encode HealthCheck as a [Value Class](https://docs.scala-lang.org/overviews/core/value-classes.html) 
* Use the `ServerStatus` `case class` for the `AllStatus`
* Allowing the user to specify the status type (to allow something different to a `String`). *I'm not fully sure if this will be supported by the macro, but it worth a try*

```scala

class HealthCheck(val nameService: String) exends AnyVal
final case class ServerStatus[A](status: A)
final case class HealthStatus[A](hc: HealthCheck, status: ServerStatus[A])
final case class AllStatus[A](all: List[HealthStatus[A]])

trait HealthCheckServiceUnary[F[_], A] {
    def setStatus[A](newStatus: HealthStatus[A]): F[Unit]

    def check[A](service: HealthCheck): F[ServerStatus[A]]
    def clearStatus(service: HealthCheck): F[Unit]

    def checkAll[A](empty: Empty.type): F[AllStatus[A]]
    def cleanAll(empty: Empty.type): F[Unit]
}
```

Reply:
I think it's useful for the user to get a confirmation about the set or clean process.  
About ServerStatus, my first try was to make it an enum type but there is a bug and it picks the first value in alphabetical order. I'll try to change the status type to an A type.

Reply:
If you use update instead of tryUpdate, you can be sure the update is made.

Reply:
![image](https://user-images.githubusercontent.com/44172755/61875723-94e39080-aeeb-11e9-9e8f-72d2a69bf5e8.png)

:'(

Reply:
![](https://media.giphy.com/media/jUwpNzg9IcyrK/source.gif)

`String` is perfect then 😅